up-streamer/tests/single_local_single_remote.rs

```diff
@@ -0,0 +1,236 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use async_broadcast::broadcast;
+use async_std::sync::{Condvar, Mutex};
+use async_std::task;
+use futures::future::join;
+use integration_test_utils::{
+    check_messages_in_order, check_send_receive_message_discrepancy, local_authority,
+    local_client_uuri, notification_from_local_client_for_remote_client,
+    notification_from_remote_client_for_local_client, publish_from_local_client_for_remote_client,
+    publish_from_remote_client_for_local_client, remote_authority_a, remote_client_uuri,
+    request_from_local_client_for_remote_client, request_from_remote_client_for_local_client,
+    reset_pause, response_from_local_client_for_remote_client,
+    response_from_remote_client_for_local_client, run_client, signal_to_pause, signal_to_resume,
+    wait_for_pause, ClientCommand, LocalClientListener, RemoteClientListener, UPClientFoo,
+};
+use log::debug;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use up_rust::{UListener, UTransport};
+use up_streamer::{Endpoint, UStreamer};
+
+const DURATION_TO_RUN_CLIENTS: u128 = 1_000;
+const SENT_MESSAGE_VEC_CAPACITY: usize = 10_000;
+
+#[async_std::test]
+async fn single_local_single_remote() {
+    // using async_broadcast to simulate communication protocol
+    let (tx_1, rx_1) = broadcast(10000);
+    let (tx_2, rx_2) = broadcast(10000);
+
+    let utransport_foo: Arc<dyn UTransport> =
+        Arc::new(UPClientFoo::new("upclient_foo", rx_1.clone(), tx_1.clone()).await);
+    let utransport_bar: Arc<dyn UTransport> =
+        Arc::new(UPClientFoo::new("upclient_bar", rx_2.clone(), tx_2.clone()).await);
+
+    // setting up streamer to bridge between "foo" and "bar"
+    let ustreamer = UStreamer::new("foo_bar_streamer", 3000);
+
+    // setting up endpoints between authorities and protocols
+    let local_endpoint = Endpoint::new("local_endpoint", local_authority(), utransport_foo.clone());
```

Comment:
Cloning the `Arc<dyn UTransport>`?

An `Endpoint` is a unique pair of `UAuthority` and `Arc<dyn UTransport>`.

We clone so that we can re-use the same `Arc<dyn UTransport>` as a part of another `Endpoint` with a different `UAuthority`.

Reply:
I mean nobody else uses the `utransport_foo` and `utransport_bar` in this function. We can still use `Arc`, but no need to clone here.

Reply:
Right. That's fair.

I suppose normally we would clone for the cases where we need to add more forwarding rules using the UTransport, but it's not needed in this case.

Reply:
Fixed!